### PR TITLE
Order details show refunded shipping lines in refund summary

### LINKED ISSUE #6714: [Order Details] Show refunded shipping lines in refund summary
⛔️ Awaiting design confirmation: p91TBi-8P0-p2#comment-10438

Refunded shipping is not displayed anywhere on the Order Details screen, nor in the refund details that you can tap through to.

> There’s an [outstanding issue on Android](https://github.com/woocommerce/woocommerce-android/issues/3853) from last year to add a line for shipping refunds within the Refunded Products section.
> – @joshheald

> The shipping cost is only referred to in the Payments section – so it seems more natural to me that we’d indicate a shipping refund somewhere here, rather than in the products section.
> maybe “Refunded: 2 products and shipping ––– $33.00”?
> – @joe-keenan

> Image
>
> I think it’s the right place but I also think that we’ll have problems with larger Dynamic Type sizes and some localizations.
> – @joshheald

For context: p91TBi-8P0-p2#comment-10327

- [ ] Include refunded shipping lines on Order Details

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #6714

### Description

Adds more info in the refund lines. 

Possible cases:
* Refunded: 1 product
* Refunded: 2 products
* Refunded: 1 product, shipping
* Refunded: 1 product, fees
* Refunded: 1 product, shipping, fees
* Refunded: shipping, fees
* Refunded: fees
* Refunded: shipping

@joe-keenan @joshheald wdyt about this approach?

### Images/gif

image
image

### CODE REVIEW

**Comment by joshheald** [comment]
> @joe-keenan @joshheald wdyt about this approach?
> 
> image

I think Android's font is more forgiving than iOS's!

Removing the word "and" helps too, even though it doesn't seem quite as polished. On iOS, the English text still reaches more than half way across the row with a comma instead of "and", so I'm still worried about truncation for localizations with longer words or in larger dynamic type sizes.

Joe is taking a look at other options too.

Another difference: on iOS we style these lines with colour, to indicate that they can be tapped to show more detail. Has Android never had that? 

Perhaps they should have a disclosure indicator instead, but perhaps the reason they don't is that it would take the amounts out of alignment in the tableview.

**Comment by joe-keenan** [comment]
> Another difference: on iOS we style these lines with colour, to indicate that they can be tapped to show more detail. Has Android never had that?

🤯 
_That’s_ why they’re pink on iOS? I’ve been looking at this screen so much and I never thought to tap those refund rows. I just thought it was calling out the negative value! 

> Perhaps they should have a disclosure indicator instead, but perhaps the reason they don't is that it would take the amounts out of alignment in the tableview.

I think an indicator of some sort would be a lot more accessible, but you’re right about it being tricky to find a spot for it. I’ll add it to my list of things to think about.

**Comment by kidinov** [comment]
@joe-keenan @joshheald 

> Another difference: on iOS we style these lines with colour, to indicate that they can be tapped to show more detail. Has Android never had that?

yep, on android it's not clickable at all. Just the text

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/details/adapter/OrderDetailRefundsLineBuilderTest.kt [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+        // WHEN
+        val result = builder.buildRefundLine(refund)
+
+        // THEN
+        assertThat(result).isEqualTo("shipping, fees")
+    }
+
+    @Test
+    fun `given refund with product, shipping and fee, when building line, then product shipping and fee returned`() {
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/details/adapter/OrderDetailRefundsLineBuilderTest.kt:154]: ❓ What do you think about writing tests that verify multiple products? 
Like 
"2 products, shipping"
"5 products, fees"
"10 products, shipping, fees"

**Review by AnirudhBhat — APPROVED** [review]
@kidinov Thank you for working on this! I've tested the changes and it works as expected. The code changes look good as well. I've left a minor comment related to unit tests and it is not blocking at all.

I read in the linked issue that the designs are not finalized for this. I'll not merge this PR as of now. Feel free to merge this if you think that the design part can be taken in a separate PR.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/details/adapter/OrderDetailRefundsLineBuilderTest.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/details/adapter/OrderDetailRefundsLineBuilderTest.kt:154]: Added more tests here - 767754993b7a87ac2563fb338585938e7ecacc08

**Comment by kidinov** [comment]
@joe-keenan @joshheald I'll merge this PR for now (it goes to a feature branch so no changes to production) and add a comment to our P2 with the changes between iOS and Android here so we can decide on what platform extra changes are needed

**Pull request merged**